internal/app/go-filecoin/connectors/storage_market/client.go

```diff
@@ -191,7 +201,7 @@ func (s *StorageClientNodeConnector) ValidatePublishedDeal(ctx context.Context,
 
 // SignProposal uses the local wallet to sign the given proposal
 func (s *StorageClientNodeConnector) SignProposal(ctx context.Context, signer address.Address, proposal market.DealProposal) (*market.ClientDealProposal, error) {
-	buf, err := encoding.Encode(proposal)
+	buf, err := cborutil.Dump(&proposal)
```

Comment:
Why this change? The encoding library should work here (tho passing a pointer looks good)? If it doesn't, we have a deeper problem that should not be skimmed over.

Reply:
It was unnecessary. The pointer was the fix. I switched it back.